Feature/1.1/#31 startup resource credits fix

brianoflondon opened the pull request:
Closes #31 with startup checks and extends test time on simulcast operation.

Github says this will only close the issue if the pull request is to the default branch.

brianoflondon committed:
Separate out construct operation

Moved the construction of the operation into own routine for use during startup

brianoflondon committed:
Use rc.get_cost to find out podping capacity during startup

brianoflondon committed:
Much longer timeout on Simulcast.

This test is hitting the condition of more than 5 posts in a single block.

Block with 4 custom_jsons in. https://hive.ausbit.dev/b/60857464#9582090d856566d327b618aed4083d5f06a07e9c

Block with 5 custom_jsons in
https://hive.ausbit.dev/b/60857458#8528d16b4d288b62c69ec1cb99d6f45446ee0e88

brianoflondon committed:
bump version

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
fixed operation_id during startup in tests

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
new test script to check the startup.

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
change startup id

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
test the podping settings check loop

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
remove settings loop test from startup

remove settings loop test from the startup test because it is tested elsewhere

brianoflondon reviewed (commented):
I think I've done the job here. Startup is streamlined and only posts one custom_json while resource credit usage is calculated by the Lighthive library.

Also added a proper test for the startup code (because it wasn't covered by tests before).

agates commented on src/podping_hivewriter/podping_hivewriter.py:
```diff
@@ -161,32 +161,31 @@ async def test_hive_resources(self):
-            logging.info(
-                f"Testing Account Resource Credits {settings.hive_operation_period}s"
+            op, size_of_json = await self.construct_operation(
+                custom_json, STARTUP_OPERATION_ID
```
Note sure why this is working but this is an invalid type... you have construct_operation() requiring "HiveOperationId" but giving it a string.

Also, STARTUP_OPERATION_ID is just set to "_startup".  Is this what you intended to set as the operation_id?

brianoflondon replied:
What I'd like is for all pings to start with pp_ or pplt_ for podping and podping livetest when we're testing.

What I really don't understand is how the Operation ID works with the GPRC or whatever that google compiler is.

So whatever the correct way to add _startup for this (now single) custom_json... that's what we should do.

agates commented on src/podping_hivewriter/podping_hivewriter.py:
```diff
@@ -161,32 +161,31 @@ async def test_hive_resources(self):
-            await self.send_notification(custom_json, STARTUP_OPERATION_ID)
+            hive_operation_id = self.operation_id + STARTUP_OPERATION_ID
+
+            await self.send_notification(custom_json, hive_operation_id)
```
Similar issue to line 164 above... str is an invalid type for send_notification.  Maybe we could change the signature to `Union[HiveOperationId, str]`?

agates commented on src/podping_hivewriter/constants.py:
```diff
@@ -1,6 +1,6 @@
 LIVETEST_OPERATION_ID = "pplt"
 PODPING_OPERATION_ID = "pp"
-STARTUP_OPERATION_ID = "ppst"
+STARTUP_OPERATION_ID = "_startup"
```
If this is intended to be a suffix to either `LIVETEST_OPERATION_ID` or `PODPING_OPERATION_ID` we should probably indicate as such.

brianoflondon replied:
Better comment elsewhere.

agates reviewed (changes requested):
Couple of small things but makes sense overall.

Commit:
poetry.lock update missed after minimum python version bump to 3.8

brianoflondon committed:
Merge remote-tracking branch 'origin/feature/1.1/#27_add-medium-reason-to-cli-interface' into feature/1.1/#31_startup_resource_credits_fix

brianoflondon committed:
move valid hive account check

Moved the check for a valid Hive account into podping.py out of podping_hivewriter.py.

This still doesn't check we have a good posting key but it is better and fails immediately if the hive account given is bad

brianoflondon committed:
Added Base64 check for hive key

Validates the Hive key for base64. Doesn't check if it can post (still working on how to do this)

brianoflondon committed:
previous base64 commit wrong, replaced with base58

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
error trapping much better

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
Allow full 15mins for simulcast test

This test deliberately forces too many simultaneous transactions on one Hive account.

It can fail when Github runs it 6 times simultaneously especially on an active pull request.

brianoflondon committed:
LINTING!!!!

Signed-off-by: Brian of London <brian@podping.org>

Revert "LINTING!!!!"

This reverts commit 1e9eeb49564deafe686e465910529282636d1d42.

automatic linting?

Signed-off-by: Brian of London <brian@podping.org>

autolint?

Signed-off-by: Brian of London <brian@podping.org>

can't get pre-commit working

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon committed:
Now does proper check pubkey to private key

brianoflondon committed:
fix line length

Signed-off-by: Brian of London <brian@podping.org>

brianoflondon commented:
I'm pretty sure the whole way startup is checked is now much more reliable and solid.

Resource credit calc is also as good as it can be.

brianoflondon committed:
fixed type hints for OperationID

brianoflondon committed:
added mypy to dev dependencies

Commit:
Remove USE_TEST_NODE from startup.  It's always false anyway.
